### PR TITLE
README.md, docs: add an introduction to parser

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 
The current README is not formal and is a bit messy. For example, section `How to use it` should be moved to a separate page.

### What is changed and how it works?
- add an introduction to parser
- move unrelated section out of README
- add docs/ for reference

### Check List 

Tests 

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by tangenta** [comment]
@gingerkidney PTAL

File: README.md [hunk 1]
```diff
@@ -5,86 +5,45 @@
...
+found you are one of the users but not listed here:
 
-File a PR to this repository. **Link the related PR in TiDB in your PR description or comment.**
+- [pingcap/tidb](https://github.com/pingcap/tidb)
+- [XiaoMi/soar](https://github.com/XiaoMi/soar)
+- [XiaoMi/Gaea](https://github.com/XiaoMi/Gaea)
+- [sql-machine-learning/sqlflow](https://github.com/sql-machine-learning/sqlflow)
 
-This PR will be reviewed, and if everything goes well, it will be merged.
+## Acknowledgments
```
Comment (zz-jason) [inline README.md]: usually, `Acknowledgments` is the last part of a README file.

File: README.md [hunk 2]
```diff
@@ -5,86 +5,45 @@
...
+- Improve performance and benchmark
+- Improve the quality of code and comments
 
-### Step 1: Make changes in your parser repository
+## Getting Help
 
-Fork this repository to your own account and commit the changes to your repository.
+- [GitHub Issue](https://github.com/pingcap/parser/issues)
+- [Stack Overflow](https://stackoverflow.com/questions/tagged/tidb)
+- [User Group (Chinese)](https://asktug.com/)
```
Comment (zz-jason) [inline README.md]: slack channel is method to get help as well.

File: README.md [hunk 3]
```diff
@@ -5,86 +5,45 @@
...
-	stmtNodes, _, err := p.Parse("select * from tbl where id = 1", "", "")
+- Highly compatible with MySQL: it supports almost all features of MySQL. For the complete details, see [parser.y](https://github.com/pingcap/parser/blob/master/parser.y) and [hintparser.y](https://github.com/pingcap/parser/blob/master/hintparser.y).
+- Extensible: adding a new syntax requires only a few lines of Yacc and Golang code changes. As an example, see [PR-680](https://github.com/pingcap/parser/pull/680/files).
+- Good performance: the parser is generated by goyacc in a bottom-up approach. It is efficient to build an AST tree with a state machine.
 
-	// 3. Use AST to do cool things.
-	fmt.Println(stmtNodes[0], err)
-}
-```
+## Documentation
```
Comment (zz-jason) [inline README.md]: maybe `How to use`?

File: README.md [hunk 4]
```diff
@@ -5,86 +5,45 @@
...
 
-See [https://godoc.org/github.com/pingcap/parser](https://godoc.org/github.com/pingcap/parser)
+Please read the [quickstart](https://github.com/pingcap/parser/blob/master/docs/quickstart.md).
 
-## How to update parser for TiDB
+## Future
```
Comment (zz-jason) [inline README.md]: maybe `Feature`? BTW, I think we should introduce this section before `Documentation`

Comment (kennytm) [reply README.md]: This section really menas "Future" as in "things to do in the future".

Comment (zz-jason) [reply README.md]: OK 😂

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**